Various Changes

DaveYognaught opened the pull request:
Bunch of minor modifications. 
Let me know of any thoughts Red. Eager to continue developing and improving the early-game experience, next PR is something pretty nostalgic and fun the new players and old should be able to appreciate to replace a currently unused Key-Item! 

A lot of this is focused on new player experience being the most common type of comments I see and something I can emphasise with being pretty recent myself. Estimated little to no impact on other players. 

**Dev Stuff:** 
- Updated gitIgnore to ignore some VS / VSCode local code. For future devs just in case they push personal settings, etc. 
Settled on using VS 2019 myself but added the VSCode folder it generated anyway. 

**Quests:**
- Changed Quit Quest confirmation message to clarify that you're able to rechallenge quests once quit.

**Pokeballs:**
- No longer the same symbol for uncaught and caught shinies. The caught shinies emoji is now more 'wham, you did it' than the scattered shiny stars. Take a lot around at what star emojis there are but I think this is the best colourful, unique one we can get. 
  This'll help new players distinguish there's a difference between the two and hopefully hover over them. 
  
**Breeding:**
- A quick change to the Breeding Key item description that makes it a clearer the importance of the hatchery and breeding earlier in the game. This Key Item only pops up once and in most cases, this'll be the one and only time a player sees this. So this one interaction we have is the only first impression we have with new players of the Hatchery. Just gives them a little more to read and understand it's breeding, it's good for attack and can even get unique pokemon from it.
Exclusively a new player thing. 

Will help address these two points i've seen on Discord:
--> "Make it more obvious Breeding increases idle attack and is key to progressing smoothly into new Regions"
--> "Breeding doesn't mention parent pokemon creating children at any point. (I think so anyway; or it's not obvious)" <-- I personally struggled with this too. One of the few issues I actually had in this game 7 days in! 

**END**
**Updated 15/11 21:04 to reflect changes**

RedSparr0w commented on docs/scripts/script.min.js:
```diff
@@ -53,6 +53,15 @@ class Evolution {
         return shiny;
     }
 }
+function ToggleVisibility(id) {
+    //alert("Toggling Visibility")
+    var x = document.getElementById(id);
+    if (x.style.display === "none") {
+        x.style.display = "grid";
+    } else {
+        x.style.display = "none";
+    }
+} 
```
Can you please revert all the changes made to the `docs/` directory, this is only updated when we push to the live website.

DaveYognaught replied:
Gotcha. Leave it with me.

RedSparr0w commented on src/components/breeding.html:
```diff
@@ -117,7 +117,7 @@ <h5 class="modal-title" id="breedingModalLabel">Day Care</h5>
                             <li class="eggSlot col-sm-4 col-md-3 col-lg-2 pokedexEntry text-nowrap" data-bind="visible: BreedingController.visible($data), style:{background: PokedexHelper.getBackgroundColors($data.name)}, class: App.game.breeding.hasFreeEggSlot() || App.game.breeding.hasFreeQueueSlot() ? '' : 'disabled'">
                                 <span style="top: 0; border-top-left-radius: 6px; border-top-right-radius: 6px;" data-bind="text: $data.name">name</span>
-                                <div data-bind="visible: App.game.party.alreadyCaughtPokemon($data.id, true)" style="position: absolute;right: 0px;top: 0px;">✨</div>
+                                <div data-bind="visible: App.game.party.alreadyCaughtPokemon($data.id, true)" style="position: absolute;right: 0px;top: 0px;">🌟</div>
```
I'd rather keep the sparkles instead of the star for these please.

DaveYognaught replied:
I'll either revert this or see what I can do to the Sparkles emoji to keep it consistent but unique somehow. Leave it with me.

RedSparr0w commented on src/components/shortcutsContainer.html:
```diff
@@ -26,6 +26,13 @@
               </button>
             </td>
           </tr>
+          <tr id="dockShortcut" data-bind="visible: Settings.getSetting('enableDockShortcut').value">
+            <td class="p-0">
+              <button class="btn btn-block btn-primary m-0" onclick="MapHelper.openShipModal()">
+                  Docks
+              </button>
+            </td>
+          </tr>
```
I've mentioned previously in the Discord, the Dock shortcut is not something to be added to the game.

DaveYognaught replied:
Totally out of the question even it's opt-in within Settings and disabled by default? I thought that was a good compromise. I'm thinking moreso along the line of mobile compatability here, as smaller screens I can actually understand a dock might be annoying to scroll up to or find.

If you say no in the end, do I revert these commits or would you be able to pick and choose your end, not used GitHub too much. I don't mind either way, btw, just thought i'd get your opinion of an actual commit to see how it would work.

RedSparr0w replied:
Unfortunately you can't just pick and choose which commits to merge in a PR, the whole PR is merged at once, so will need to be reverted on your end.

DaveYognaught replied:
Gotcha, no problem. So just to clarify, sorry for being a pest, that's definite no? 
No issue to revert at all, all seperate commits. :)

RedSparr0w commented on src/scripts/items/BattleItem.ts:
```diff
@@ -13,9 +13,9 @@ class BattleItem extends Item {
-ItemList['Token_collector'] = new BattleItem(GameConstants.BattleItemType.Token_collector, '+50% Bonus to dungeon tokens gained', 1000, undefined, 'Token Collector');
-ItemList['Item_magnet']     = new BattleItem(GameConstants.BattleItemType.Item_magnet, '+50% Chance of gaining an extra item', 1500, undefined, 'Item Magnet');
-ItemList['Lucky_incense']   = new BattleItem(GameConstants.BattleItemType.Lucky_incense, '+50% Bonus to money gained', 2000, undefined, 'Lucky Incense');
+ItemList['xAttack'] = new BattleItem(GameConstants.BattleItemType.xAttack, '+50% Bonus to Pokémon attack for 30 seconds', 600);
+ItemList['xClick'] = new BattleItem(GameConstants.BattleItemType.xClick, '+50% Bonus to click attack for 30 seconds for 30 seconds for 30 seconds', 400);
+ItemList['Lucky_egg'] = new BattleItem(GameConstants.BattleItemType.Lucky_egg, '+50% Bonus to experience gained for 30 seconds', 800, undefined, 'Lucky Egg');
+ItemList['Token_collector'] = new BattleItem(GameConstants.BattleItemType.Token_collector, '+50% Bonus to dungeon tokens gained for 30 seconds', 1000, undefined, 'Token Collector');
+ItemList['Item_magnet'] = new BattleItem(GameConstants.BattleItemType.Item_magnet, '+50% Chance of gaining an extra item for 30 seconds', 1500, undefined, 'Item Magnet');
+ItemList['Lucky_incense'] = new BattleItem(GameConstants.BattleItemType.Lucky_incense, '+50% Bonus to money gained for 30 seconds', 2000, undefined, 'Lucky Incense');
```
These extended descriptions feel unnecessary.

DaveYognaught replied:
I would agree if not for the fact it was actually a recurring comment on the Discord players weren't even aware they could stack battle items pre the quantity change. I don't think it hurts to clarify the duration since otherwise it's left to experimentation. Let's new players calculate the time they'd gain, etc. 

Whatever you prefer.

RedSparr0w replied:
I think it would be better to display this information somewhere else rather than on the tooltip for every battle item.

DaveYognaught replied:
No problem, I have another idea in that case, will revert and come back to this in a seperate PR.

RedSparr0w commented on src/scripts/keyItems/KeyItems.ts:
```diff
@@ -44,7 +44,7 @@ class KeyItems implements Feature {
             // TODO obtain somewhere at the start
             new KeyItem(KeyItems.KeyItem.Holo_caster, 'A device that allows users to receive and view hologram clips at any time. It’s also used to chat with others', undefined, undefined, undefined, 'Holo Caster'),
-            new KeyItem(KeyItems.KeyItem.Mystery_egg, 'A mysterious Egg obtained from Mr. Pokémon. This allows you to use the Pokémon Day Care', function () {
+            new KeyItem(KeyItems.KeyItem.Mystery_egg, 'A mysterious Egg obtained from Mr. Pokémon. This allows you to use the Pokémon Day Care. Breeding helps improve your Pokémons attack and some Pokémon can only be found through it too!', function () {
```
```suggestion
            new KeyItem(KeyItems.KeyItem.Mystery_egg, 'A mysterious Egg obtained from Mr. Pokémon. This allows you to use the Pokémon Day Care to help improve your Pokémons attack, some baby Pokémon can only be found through breeding too!', function () {
```

RedSparr0w commented on src/scripts/quests/Quest.ts:
```diff
@@ -56,10 +56,11 @@ abstract class Quest {
 
     quit(shouldConfirm = false) {
-        if (shouldConfirm && !confirm('Are you sure you want to quit this quest?!')) {
+        if (shouldConfirm && !confirm('Are you sure? You can restart the quest later but you will lose all progress!')) {
```
```suggestion
        if (shouldConfirm && !confirm('Are you sure?\nYou can start the quest again later but you will lose all progress!')) {
```

RedSparr0w commented on src/scripts/quests/Quest.ts:
```diff
@@ -56,10 +56,11 @@ abstract class Quest {
             return false;
         }
         this.initial(null);
+        this.isCompleted?.dispose?.();
```
This shouldn't be here, was causing problems with quitting then restarting quests.
```suggestion
```

DaveYognaught commented:
Should be good now Red.

- Reverted the /docs changes indefinitely. This'll need to be a local .git ignore in the future, not on a .gitignore level.
--> If you don't mind me asking, how do you test your branches locally I had put it in /docs so I could run it locally. If I continue to do so i'll need to ignore it but do you run it from a different path? 
- Reverted Docks changes indefinitely. 
- Reverted Battle Item changes for a better idea i've had for a future PR.   
 
Quests:
- Amended to as suggested. 

Breeding:
- Amended to as suggested. 

Pokeballs:
- Keeping the current Sparkles for consistency, however, i've now dulled the icon for already caught pokemon within the Pokeballs tab. At 80% greyscale opacity. Still has enough colour but isn't fully grey or not dulled enough, I think this is a good balance.

![Screenshot_1](https://user-images.githubusercontent.com/58609098/99200844-ae26e500-279f-11eb-8029-cfad2d95b9a6.png)

Kinda reminds me of when you could rub and clean up your Pokemon Badges to get them shiny again! So I actually really like this a lot more than before. The shiny icon stays shiny for other modals, ie, breeding, dungeon or pokedex. It's only on that pokeball tab i've changed.

RedSparr0w reviewed (approved):
👍 Looks good, thanks for the PR!
